### PR TITLE
ISSUE-4478: remove setting `parallel_read_threads`

### LINKED ISSUE #4478: Consider deprecate `parallel_read_threads`  setting
**Summary**

- `parallel_read_threads` could be choosen adaptively , i.e. ` ceil(number_of_blocks / number_cpus) `

### PR DESCRIPTION
**Pull request by dantengsky**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

~~deprecate~~ remove setting `parallel_read_threads`

## Changelog

- Improvement
- Not for changelog (changelog entry is not required)

## Related Issues

Fixes #4478

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by dantengsky** [comment]
@wubx setting `parallel_read_threads` is being removed

**Comment by dantengsky** [comment]
@zhang2014 After the "new processor" deprecated the current processor, I'd like to cleanup those legacy codes.

File: query/src/storages/fuse/operations/read.rs [hunk 1]
```diff
@@ -42,14 +42,19 @@ impl FuseTable {
...
     ) -> Result<SendableDataBlockStream> {
         let block_reader = self.create_block_reader(&ctx, push_downs)?;
 
-        let bite_size = ctx.get_settings().get_parallel_read_threads()?;
-        let iter = std::iter::from_fn(move || match ctx.clone().try_get_partitions(bite_size) {
-            Err(_) => None,
-            Ok(parts) if parts.is_empty() => None,
-            Ok(parts) => Some(parts),
+        let max_threads = ctx.get_settings().get_max_threads()?;
+        let current_block_reading = num::Integer::div_ceil(&(num_parts as u64), &max_threads);
```
Comment (zhang2014) [inline query/src/storages/fuse/operations/read.rs:50]: All part is concurrent? When `max_ threads = 8` and  `parts.len() = 16`, we will create 8 ✖️ SendableDataBlockStream, and read two parts in each stream?

Comment (dantengsky) [reply query/src/storages/fuse/operations/read.rs:50]: yeah, seems too aggressive & and has no significant effect at least in CPU bounded tasks either.
let me refine it

Comment (dantengsky) [reply query/src/storages/fuse/operations/read.rs]: @zhang2014  PTAL

**Pull request merged**